etna/records/tests/test_views.py

```diff
@@ -722,3 +725,97 @@ def test_invalid_response_from_kong_raises_404(self):
+        self.back_to_search_url_timestamp = timezone.now()
+
+    @responses.activate
+    def test_back_to_search_render_with_catalogue_search_within_expiry(self):
+        """navigation to record details from previous search (session is set since its coming from search catalogue)"""
+
+        search_url_gen_html_resp = "%2Fsearch%2Fcatalogue%2F%3Fsort_by%3Dtitle%26q%3Dlondon%26filter_keyword%3Dpaper%26level%3DItem%26collection%3DADM%26collection%3DBT%26closure%3DOpen%2BDocument%252C%2BOpen%2BDescription%26opening_start_date_0%3D%26opening_start_date_1%3D%26opening_start_date_2%3D1900%26opening_end_date_0%3D%26opening_end_date_1%3D%26opening_end_date_2%3D2020%26per_page%3D20%26sort_order%3Dasc%26display%3Dlist%26page%3D2%26group%3Dtna"
+
+        session = self.client.session
+        session["back_to_search_url"] = search_url_gen_html_resp
+        session[
+            "back_to_search_url_timestamp"
+        ] = self.back_to_search_url_timestamp.isoformat()
+        session.save()
+
+        response = self.client.get(self.record_detail_url)
+
+        expected_button_link_gen_value = self.expected_button_link_gen_value_fmt.format(
+            back_to_search_url=search_url_gen_html_resp,
+        )
+        self.assertContains(response, expected_button_link_gen_value)
+
+    @responses.activate
+    def test_back_to_search_render_with_catalogue_search_beyond_expiry(self):
+        """navigation to record details from previous search (session is set since its coming from search catalogue)"""
+
+        search_url_gen_html_resp = "/search/featured/"
+
+        session = self.client.session
+        session["back_to_search_url"] = search_url_gen_html_resp
+        # set time behind the setup value for expiry
+        session["back_to_search_url_timestamp"] = (
+            self.back_to_search_url_timestamp - SEARCH_URL_RETAIN_DELTA
+        ).isoformat()
+        session.save()
+
+        response = self.client.get(self.record_detail_url)
+
+        expected_button_link_gen_value = self.expected_button_link_gen_value_fmt.format(
+            back_to_search_url=search_url_gen_html_resp,
+        )
+        self.assertContains(response, expected_button_link_gen_value)
+
+    @responses.activate
+    def test_new_search_render_without_session(self):
+        """navigation to record details without previous search (session is not set since its not coming from search)"""
+
+        new_search_url = reverse("search-featured")
+
+        response = self.client.get(self.record_detail_url)
+
+        expected_button_link_gen_value = self.expected_button_link_gen_value_fmt.format(
+            back_to_search_url=new_search_url
+        )
+        self.assertContains(response, expected_button_link_gen_value)
+
+    @responses.activate
+    def test_new_search_render_with_session(self):
+        """navigation to record details from previous search (session is set since its coming from search featuerd, but without query)"""
```

Comment:
Not a blocking thing, but these docstrings should really do a better job of outlining the expectations of the test - not just the scenario they are testing.

Reply:
I came up with this "Test covers navigation to record details without a previous search (session is not set since its not coming from search)". Happy to change it otherwise.